deepspeed/runtime/activation_checkpointing/checkpointing.py

```diff
@@ -427,15 +430,18 @@ def get_partitioned_activations_for_backward(args, inputs, contiguous_checkpoint
     global contiguous_size_buffers, size_offsets
 
     new_args = []
+    num_non_fp_tensors = 0
     for i, (arg, inp) in enumerate(zip(args, inputs)):
         size = torch.tensor(arg.size()) if torch.is_tensor(arg) else None
         if not is_activation_to_checkpoint(arg):
             new_args.append(arg)
             new_args.append(size)
+            num_non_fp_tensors += 1
             continue
 
         arg.data = inp.data
         new_args.append(arg)
+        i -= num_non_fp_tensors

```

Comment:
Another approach for minimal changes would: 
```python
434    for arg_index, (arg, inp) in ....
         ....
444         i = arg_index - num_non_fp_tensors
```

What do you think?

Reply:
I think this way looks better.

Reply:
Great. I will wait for you to make the changes. Thanks.